### DISCUSSION TITLE
inconsistencies between the document and the API behavior

### DISCUSSION THREAD

**Discussion by jpnjfk-sas** [unicode-org/message-format-wg#522] [General]
Hi all,
I joined this WG a couple of weeks. If have something out of the ordinal manner in this WG, let me know please.
Using the APIs, I found a few inconsistencies between the documents ([syntax.md](https://github.com/unicode-org/message-format-wg/blob/main/spec/syntax.md), [message.abnf](https://github.com/unicode-org/message-format-wg/blob/main/spec/message.abnf)) and the API behavior as below.

1. The complex-message which is enclosed by the "{{" and "}}" causes the parse error.
  When the "{{" and "}}" are removed, the API works.
2. The non-terminal local causes the parse error with the terminal "local" in the complex-message. The API expects "let" instead of "local".
3. The API doesn't accept the non-terminal quoted-pattern for the non-terminal variant.
The API accepts the non-terminal expression instead of quoted-pattern.

Which one is correct? The documents or the API behavior?
If the API behavior is correct, I'm willing to make changes in the documents and create a pull-request.

**Comment by eemeli** [comment]
Could you clarify what you mean by "the API behavior"? If it's the ICU4J or JS `messageformat` implementation, those have not yet been updated to account for recent spec changes.

**Reply by jpnjfk-sas** [reply]
It's icu4j com.ibm.icu.message2.MessageFormatter.builder().setPattern(message).build().

**Reply by eemeli** [reply]
That API has not yet been updated to reflect spec changes.

**Reply by jpnjfk-sas** [reply]
Thank you!
